### PR TITLE
Add ynh_add_config helper

### PR DESCRIPTION
**Pull request by alexAubin**
## The problem

Too much madness about ynh_replace_string all over the place, lack of consistent/simple template format/helpers

## Solution

Add yalh's experimental helper ynh_add_config into the core

## PR Status

Yolocomitted, but done I guess ? 

## How to test

Uuuh idk

### CODE REVIEW

File: data/helpers.d/utils [hunk 1]
```diff
@@ -263,6 +263,147 @@ ynh_local_curl () {
...
+#
+# The helper will replace the following keywords with global variables
+# that should be defined before calling this helper :
+#   __PATH__             by  $path_url
+#   __NAME__            by $app
+#   __NAMETOCHANGE__    by $app
+#   __USER__            by $app
+#   __FINALPATH__       by $final_path
+#   __PHPVERSION__      by $YNH_PHP_VERSION
+#
```
Comment (ericgaspar) [inline data/helpers.d/utils:345]:
```suggestion
#   __YNH_NODE_LOAD_PATH__"    by $ynh_node_load_PATH
```

**Review by ericgaspar — COMMENTED** [review]
as it is also in the script... idk

File: data/helpers.d/utils [hunk 2]
```diff
@@ -263,6 +263,147 @@ ynh_local_curl () {
...
+# The helper will replace the following keywords with global variables
+# that should be defined before calling this helper :
+#   __PATH__             by  $path_url
+#   __NAME__            by $app
+#   __NAMETOCHANGE__    by $app
+#   __USER__            by $app
+#   __FINALPATH__       by $final_path
+#   __PHPVERSION__      by $YNH_PHP_VERSION
+#   __YNH_NODE_LOAD_PATH__"    by $ynh_node_load_PATH
```
Comment (ericgaspar) [inline data/helpers.d/utils:345]:
```suggestion
#   __PATH__                by $path_url
#   __NAME__                by $app
#   __NAMETOCHANGE__        by $app
#   __USER__                by $app
#   __FINALPATH__           by $final_path
#   __PHPVERSION__          by $YNH_PHP_VERSION
#   __YNH_NODE_LOAD_PATH__  by $ynh_node_load_PATH
#
```

File: data/helpers.d/utils [hunk 3]
```diff
@@ -263,6 +263,147 @@ ynh_local_curl () {
...
+# The helper will use the template $template to generate a config file
+# $destination by replacing the following keywords with global variables
+# that should be defined before calling this helper :
+#   __PATH__             by  $path_url
+#   __NAME__            by $app
+#   __NAMETOCHANGE__    by $app
+#   __USER__            by $app
+#   __FINALPATH__       by $final_path
+#   __PHPVERSION__      by $YNH_PHP_VERSION
+#
```
Comment (ericgaspar) [inline data/helpers.d/utils:288]:
```suggestion
#   __PATH__                by $path_url
#   __NAME__                by $app
#   __NAMETOCHANGE__        by $app
#   __USER__                by $app
#   __FINALPATH__           by $final_path
#   __PHPVERSION__          by $YNH_PHP_VERSION
#   __YNH_NODE_LOAD_PATH__  by $ynh_node_load_PATH
#
```

**Review by ericgaspar — CHANGES_REQUESTED** [review]
Sorry for the review en deux temps :s

**Review by zamentur — APPROVED** [review]
LGTM

**Pull request merged**